.github/workflows/deploy.yaml

```diff
@@ -0,0 +1,29 @@
+name: deploy-test
+
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches:
+    - main
+
+jobs:
+  deploy-basic:
+    runs-on: ubuntu-latest
+    steps:
+
+    - uses: actions/checkout@v3
+
+    - uses: actions/setup-go@v4
+      with:
+        go-version-file: go.mod
+
+    - name: Run basic unit tests
```

Comment:
calling this a "unit" test would be confusing to me if  all I saw was the name of the step and not what was actually happening